packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -152,17 +153,35 @@ const getContextMenuStyles = stylesFactory((theme: GrafanaTheme) => {
 export const ContextMenu: React.FC<ContextMenuProps> = React.memo(({ x, y, onClose, items, renderHeader }) => {
   const theme = useContext(ThemeContext);
   const menuRef = useRef(null);
+  const [positionStyles, setPositionStyles] = useState(null);
+
+  useEffect(() => {
+    if (menuRef) {
+      const rect = menuRef.current.getBoundingClientRect();
```

Comment:
If we are getting measurements shouldn't we use `useLayoutEffect`?

Reply:
yes